### ISSUE TITLE
feat: Detect JWT `alg:none` attack

### ISSUE THREAD

**Issue by S0obi** [coreruleset/coreruleset#4662]
# feat: Detect JWT `alg:none` attack

### Motivation

JWT tokens are everywhere in modern web apps, but CRS has no offensive JWT detection at all. Interestingly, the format is already *known* to the ruleset — `REQUEST-942-APPLICATION-ATTACK-SQLI.conf` (~line 1361) deliberately excludes `ey`-prefixed strings from SQLi checks to avoid false positives on bearer tokens. It feels like a natural next step to turn that awareness into actual detection.

The most actionable attack to start with is **`alg:none`** — where an attacker strips the token signature and sets the algorithm to `"none"` to bypass verification. The signature is stable and unambiguous, and the false positive risk is very low. Feels like a PL1 candidate.

### Proposed solution

Decode the JWT header segment (base64url, always starts with `eyJ`) and match `"alg":"none"` in the decoded JSON.

The decoded attack header looks like:
```
{"alg":"none","typ":"JWT"}
```

Open questions I'd love input on:
- Should this live in a new `REQUEST-945-APPLICATION-ATTACK-JWT.conf`, or is `REQUEST-934-APPLICATION-ATTACK-GENERIC.conf` a better fit given it's a single rule for now?
- Any FP scenarios worth considering? The `alg:none` string after base64 decoding seems quite specific, but happy to be corrected.

**Comment by airween** [comment]
Hi @S0obi,

thanks for bringing this up, I think prevent JWT FP's would be a great idea.

My thoughts about your questions:

>  * Should this live in a new `REQUEST-945-APPLICATION-ATTACK-JWT.conf`, or is `REQUEST-934-APPLICATION-ATTACK-GENERIC.conf` a better fit given it's a single rule for now?

I don't think we need to create a new file for this. Subset 934 would be a perfect place, or `REQUEST-921-PROTOCOL-ATTACK.conf`. Or may be a plugin, first...?

> * Any FP scenarios worth considering? The `alg:none` string after base64 decoding seems quite specific, but happy to be corrected.

As I know JWT is used only in `Authorization` header, so I don't mind this could trigger too much FP's.
